core/src/main/java/org/elasticsearch/index/query/GeoDistanceRangeQueryParser.java

```diff
@@ -205,20 +202,20 @@ public GeoDistanceRangeQueryBuilder fromXContent(QueryParseContext parseContext)
             queryBuilder.queryName(queryName);
         }
 
-        if (point != null) {
-            queryBuilder.point(point.lat(), point.lon());
-        }
-
-        if (geohash != null) {
-            queryBuilder.geohash(geohash);
-        }
-
         if (vFrom != null) {
-            queryBuilder.from(vFrom);
+            if (vFrom instanceof Number) {
+                queryBuilder.from((Number) vFrom);
+            } else {
+                queryBuilder.from((String) vFrom);
+            }
         }
 
         if (vTo != null) {
-            queryBuilder.to(vTo);
+            if (vTo instanceof Number) {
```

Comment:
you can do this simpler and just call `vTo.toString()` this should work just fine?